examples/codenames/Codenames_RL.ipynb

```diff
@@ -0,0 +1,12525 @@
+            "                                 Dload  Upload   Total   Spent    Left  Speed\n",
+            "100  2481  100  2481    0     0   6941      0 --:--:-- --:--:-- --:--:--  6949\n",
+            "  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n",
+            "                                 Dload  Upload   Total   Spent    Left  Speed\n",
+            "100 4749k  100 4749k    0     0  5151k      0 --:--:-- --:--:-- --:--:-- 5150k\n"
+          ]
+        }
+      ],
+      "source": [
+        "# Download the 400 original codenames words and a dictionary to ensure that the clue is not outside of this dictionary\n",
+        "import requests\n",
+        "codenames_words_path = \"https://raw.githubusercontent.com/OpenPipe/ART/refs/heads/main/examples/codenames/codenames_words.txt\"\n",
+        "dictionary_path = \"https://raw.githubusercontent.com/OpenPipe/ART/refs/heads/main/examples/codenames/dictionary.txt\"\n",
+        "CODENAMES_WORDS = requests.get(codenames_words_path).json()\n",
+        "DICTIONARY = requests.get(dictionary_path).json()"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "we0PhGpVoGmR",
+        "outputId": "fc39772e-7f84-4dc4-8915-ff1125636b2f"
+      },
+      "outputs": [],
+      "source": [
+        "# Basic imports to get us running\n",
+        "# ───────────────────────────────  Imports  ────────────────────────────────\n",
+        "import asyncio\n",
+        "import copy\n",
+        "import logging\n",
+        "import random\n",
+        "import re\n",
+        "import time\n",
+        "import uuid\n",
+        "from dataclasses import dataclass\n",
+        "from pathlib import Path\n",
+        "from typing import Dict, List, Tuple\n",
+        "\n",
+        "from openai import AsyncOpenAI\n",
+        "from tqdm.asyncio import tqdm_asyncio\n",
+        "\n",
+        "import art\n",
+        "from openpipe.client import AsyncOpenPipe"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "jSlnMfuvoK0O"
+      },
+      "outputs": [],
+      "source": [
+        "# ───────────────────────────────  Config  ──────────────────────────────\n",
+        "NUM_STEPS = 140\n",
+        "NUM_GAMES_PER_STEP = 10\n",
+        "NUM_ROLLOUTS_PER_GAME = 20\n",
+        "MODEL_NAME = \"codenames_model_op_3\"\n",
+        "BASE_MODEL_NAME = \"Qwen/Qwen2.5-3B-Instruct\"\n",
+        "VAL_BENCHMARK_MODEL = \"gpt-4.1-nano\"\n",
+        "NUM_VALIDATION_GAMES = 50\n",
+        "LEARNING_RATE = 5e-6\n",
+        "BETA = 0.0\n",
+        "DEBUG = False\n",
+        "\n",
+        "# — reward constants —\n",
+        "# The following is the weightage for different kinds of outcomes from a game, to determine the reward\n",
+        "MARGIN_REWARD_DICT = {\n",
+        "    \"turn_limit\": -100,\n",
```

Comment:
Would you mind adding small inline comments describing what each of these terms means?